app/src/main/java/de/othr/plantico/ui/search/SearchableActivity.kt

```diff
@@ -0,0 +1,83 @@
+import de.othr.plantico.TestViewModelFactory
+import de.othr.plantico.database.PlantApplication
+import de.othr.plantico.database.entities.Plant
+import de.othr.plantico.databinding.ActivitySearchBinding
+import de.othr.plantico.ui.homescreen.PlantAdapter
+
+class SearchableActivity: AppCompatActivity() {
+
+    private lateinit var binding: ActivitySearchBinding
+    private val testViewModel: TestViewModel by viewModels {
+        TestViewModelFactory((application as PlantApplication).repository)
+    }
+
+    var adapter: ListAdapter? = null
+    var allPlants: List<Plant> = ArrayList()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySearchBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        val adapter = PlantAdapter()
+        binding.recyclerviewSearchedPlants.adapter = adapter
+        binding.recyclerviewSearchedPlants.layoutManager = LinearLayoutManager(this)
+
+        testViewModel.allPlants.observe(this) { plants ->
+            plants.let {
+                // Set the plants list which should get searched
+                allPlants = it
+                // Set the Initial plants list in the adapter
+                // TODO: Maybe change this to frequently visited plants later on?
+                adapter.submitList(it)
+                binding.numberResultsFound.text = getString(R.string.results_found, it.size)
+                if(it.isNotEmpty()){
+                    binding.recyclerviewSearchedPlants.visibility= View.VISIBLE
+                    binding.noResultsFound.visibility= View.GONE
+                }
+            }
+        }
+
+        // Register Listener
+        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                val results = searchForPlantsInList(allPlants, query)
+                if(results.isEmpty()) {
+                    binding.recyclerviewSearchedPlants.visibility= View.GONE
+                    binding.noResultsFound.visibility= View.VISIBLE
+                }
+                adapter.submitList(
+                    results
+                )
+                binding.numberResultsFound.text = getString(R.string.results_found, results.size)
+                return false
+            }
+            override fun onQueryTextChange(newText: String): Boolean {
+                val results = searchForPlantsInList(allPlants, newText)
+                adapter.submitList(
+                    results
+                )
+                binding.numberResultsFound.text = getString(R.string.results_found, results.size)
+                return false
+            }
+        })
+    }
+
+    //TODO: Maybe change this to Filtering in database instead of accessing all plants?
+    //Search for plants that contain the query as a substring. Not case-sensitive!
```

Comment:
Yeah, this could be an interesting thing to do. It should also only be a question of creating a new query on the Repository. In general, it is also the idea, to filter or manage your data in the ViewModel. So this would also be a thing that would in general improve the code quality. But I would do it as an improvement once more attributes are available.
So leave it for now and maybe do it at a later time.